### ISSUE TITLE
Empty bindings.json file breaks build

### ISSUE THREAD

**Issue by lucsoft** [littledivy/deno_bindgen#30]
- [ ] When deno_bindgen errors out it creates a empty bindings.json which should be fixed
- [ ] Marko should check before if the file is a valid json if not recreate it

**Comment by littledivy** [comment]
@lucsoft I believe this was fixed in #28 
Does this still happen if you use the CLI and Rust crate from the `main` branch?

**Comment by lucsoft** [comment]
@littledivy yes it still happens and I'm on main

**Comment by littledivy** [comment]
Ah okay. Just to confirm this is how you updated to `main`? :-)

```toml
# Cargo.toml
[dependencies]
deno_bindgen = { git = "https://github.com/littledivy/deno_bindgen", rev = "60a6f8a" }
```

```bash
deno install -n deno_bindgen -f -A --unstable https://raw.githubusercontent.com/littledivy/deno_bindgen/main/cli.ts
```

**Comment by lucsoft** [comment]
Can confirm this still happens
i think it's a race condition or something

i can't reproduce it consistently sadly

**Comment by littledivy** [comment]
Hmm I can't reproduce it either. 
There might be an issue with a `bindings.json` from previous version conflicting with `main`. Try `rm bindings.json` and see if you can reproduce it anymore. 
Also a console output of the issue you're facing will be helpful.

**Comment by lucsoft** [comment]
i deleted it and it worked for some time but after some saves it breaks

and you get
called `Result::unwrap()` on an `Err` value: Error("EOF while parsing a value", line: 1, column: 0)rust-analyzermacro-error and the json file is empty

**Comment by lucsoft** [comment]
mostly when i make bigger changes like renaming the exported function

**Comment by littledivy** [comment]
I've added error handling for broken bindings.json. @lucsoft Can you still reproduce from latest main (update both crate and CLI)?

**Comment by littledivy** [comment]
Released in 0.3.0
